src/Test.UnitTests.BinSkim.Driver/CompilerDataLoggerTests.cs

```diff
@@ -622,6 +625,135 @@ private void LogExceptionEvents(List<ExecutionException> exceptions, CompilerDat
             return sb;
         }
 
+        private StringBuilder ValidateChunkedEvents(string expectedEventName,
+                                                    List<ITelemetry> telemetryGeneratedEvents,
+                                                    string expectedChunkedContent,
+                                                    int chunkNumber)
+        {
+            var sb = new StringBuilder();
+            string assemblyEventsId = string.Empty;
+            string commandLineEventsId = string.Empty;
+
+            var compilerEvents = new List<EventTelemetry>();
+            var assemblyReferencesEvents = new List<EventTelemetry>();
+            var commandLineEvents = new List<EventTelemetry>();
+            var summaryEvents = new List<EventTelemetry>();
+
+            foreach (EventTelemetry telemetryEvent in telemetryGeneratedEvents)
+            {
+                switch (telemetryEvent.Name)
+                {
+                    case CompilerDataLogger.CompilerEventName:
+                        compilerEvents.Add(telemetryEvent);
+
+                        if (commandLineEventsId.Equals(string.Empty) && telemetryEvent.Properties.ContainsKey("commandLineId"))
```

Comment:
>telemetryEvent [](http://example.com/codeflow?start=72&length=14)

You perform a map look up twice in your code, once for ContainsKey, the second to actually retrieve the value.

Try using a TryGetValue pattern instead. You can also inline a variable for this to assign to using new C# syntax:

if (commandLineEvents.IdEquals(string.Empty) &&
    telemetryEventProperties.TryGetValue("commandLineId", out string eventId))
{
    commandLineEventsId = eventId;
}